rpc/client.go

```diff
@@ -25,72 +24,91 @@ type UploadReportOptions struct {
 
 func UploadReport(ctx context.Context, opts UploadReportOptions) error {
 	if err := opts.convertAndUploadArtifacts(ctx); err != nil {
-		return errors.Wrap(err, "problem uploading tests for report")
+		return errors.Wrap(err, "uploading tests for report")
 	}
 	return errors.Wrap(uploadTests(ctx, gopb.NewCedarPerformanceMetricsClient(opts.ClientConn), opts.Report, opts.Report.Tests, opts.DryRun),
-		"problem uploading tests for report")
+		"uploading tests for report")
 }
 
 func (opts *UploadReportOptions) convertAndUploadArtifacts(ctx context.Context) error {
-	jobQueue := queue.NewLocalLimitedSize(runtime.NumCPU(), len(opts.Report.Tests)*2)
+	var wg sync.WaitGroup
+	catcher := grip.NewBasicCatcher()
+	testChan := make(chan poplar.Test)
```

Comment:
Should this have a small buffer size so that the producer is not blocked waiting for a consumer in the case of `!SerializeUpload`?